hypothesis-python/tests/cover/test_functions.py

```diff
@@ -125,3 +129,61 @@ def test_functions_strategy_with_kwonly_args(f):
         f(1, 2)
     f(1, kwonly_arg=2)
     f(kwonly_arg=2, arg=1)
+
+
+def pure_func(arg1, arg2):
+    ...
+
+
+@given(
+    f=functions(like=pure_func, returns=integers(), pure=True),
+    arg1=integers(),
+    arg2=integers()
+)
+def test_functions_pure_with_same_args(f, arg1, arg2):
+    assert f(arg1, arg2) == f(arg1, arg2)
+
+
+@given(
+    f=functions(like=pure_func, returns=integers(), pure=True),
+    arg1=integers(),
+    arg2=integers()
+)
+def test_functions_pure_with_different_args(f, arg1, arg2):
+    r1 = f(arg1, arg2)
+    r2 = f(arg2, arg1)
+    assume(r1 != r2)
+    assert r1 != r2
+
+
+@given(
+    f1=functions(like=pure_func, returns=integers(), pure=True),
+    f2=functions(like=pure_func, returns=integers(), pure=True)
+)
+def test_functions_pure_two_functions_different_args_same_result(f1, f2):
```

Comment:
With "any integer" allowed as the return strategy, this is very unlikely to be satisfied (by anything less than our duplication-increasing mutator pass, anyway).

I'd probably just delete this test and `test_functions_pure_two_functions_same_args_same_result` below, as they must trivially be satisfied by e.g. the default returns strategy of `none()`.